src/test/SDL_test_memory.c

```diff
@@ -67,6 +67,9 @@ static SDL_free_func SDL_free_orig = NULL;
 static int s_previous_allocations = 0;
 static SDL_tracked_allocation *s_tracked_allocations[256];
 static SDL_bool s_randfill_allocations = SDL_FALSE;
+static SDL_AtomicInt s_lock;
+#define LOCK_ALLOCATOR() do {} while (!SDL_AtomicCompareAndSwap(&s_lock, 0, 1))
```

Comment:
There's a definite possibility of thread starvation here. You should add SDL_CPUPauseInstruction() to the body of your loop.